src/test/java/com/chikli/hudson/plugin/naginator/ProgressiveDelayTest.java

```diff
@@ -52,6 +63,49 @@ public void testComputeScheduleDelayNoMax() {
                 progressiveDelay.computeScheduleDelay(createBuild(true, createBuild(true, createBuild(false, null)))));
     }
 
+    @Issue("JENKINS-43803")
+    @Test
+    public void testBuildsRunAlternately() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+        p.getBuildersList().add(new FailureBuilder());
+
+        FreeStyleBuild buildA1 = p.scheduleBuild2(0).get();
+        assertTrue(NaginatorRetryAction.scheduleBuild(
+            buildA1,
+            0,  // delay
+            1,  // retyCount
+            120   // maxRetryCount
+        ));
+        j.waitUntilNoActivity();
+        FreeStyleBuild buildA2 = p.getLastBuild();
+
+        p.scheduleBuild2(0).get();     // buildB1
+
+        assertTrue(NaginatorRetryAction.scheduleBuild(
+            buildA2,
+            0,  // delay
+            2,  // retyCount
+            120   // maxRetryCount
+        ));
+
+        j.waitUntilNoActivity();
+        FreeStyleBuild buildA3 = p.getLastBuild();
+
+        ProgressiveDelay delay = new ProgressiveDelay(300, 0);
+        assertEquals(
+            Arrays.asList(
+                300,
+                900,
+                1800
+            ),
+            Arrays.asList(
+                delay.computeScheduleDelay(buildA1),
+                delay.computeScheduleDelay(buildA2),
+                delay.computeScheduleDelay(buildA3)
+            )
+        );
```

Comment:
This fails without the fix like this:
https://ci.jenkins.io/job/Plugins/job/naginator-plugin/view/change-requests/job/PR-46/2/testReport/junit/com.chikli.hudson.plugin.naginator/ProgressiveDelayTest/testBuildsRunAlternately/
```
expected:<[300, 900, 1800]> but was:<[300, 900, 900]>
```